src/Microsoft.Azure.WebJobs.Host/InvocationFilterAttribute.cs

```diff
@@ -0,0 +1,39 @@
+﻿// InvocationFilterAttribute.cs
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// This is the first iteration for the Invocation Filter Attribute
+    /// </summary>
+    [CLSCompliant(false)]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = true, AllowMultiple = false)]
+    public abstract class InvocationFilterAttribute : Attribute
```

Comment:
The idea was that this was the base class for all invocation purposes. Currently it is possible to access instance methods that are able to be accessed in both OnExecuting and OnExecuted. The TestFailingFilter invocation test reflects this happening.